lib/src/serializable.dart

```diff
@@ -0,0 +1,224 @@
+}
+
+class SerializableEvent<T> extends Event<T> {
+  String _eventKey;
+
+  SerializableEvent(this._eventKey, DispatchKey dispatchKey)
+      : super(dispatchKey);
+
+  String get eventKey => _eventKey;
+}
+
+abstract class SerializableEvents {
+  List<SerializableEvent> get allEvents;
+}
+
+abstract class SerializableModule extends Module {
+  SerializableModule() {
+    SerializableBus.sharedBus.registerModule(this);
+  }
+
+  @override
+  SerializableEvents get events => null;
+
+  String get serializableKey => null;
+}
+
+class _ModuleRegistration extends Object with Disposable {
+  final SerializableModule module;
+
+  _ModuleRegistration(this.module);
+
+  @override
+  void manageStreamSubscription(StreamSubscription subscription) {
+    super.manageStreamSubscription(subscription);
+  }
+}
+
+class SerializableBus {
+  static final SerializableBus sharedBus = new SerializableBus();
+
+  Bridge _bridge;
+  final Logger _logger = new Logger('Serializable Bus');
+  Map<String, _ModuleRegistration> _moduleRegistrations =
+      <String, _ModuleRegistration>{};
+
+  Bridge get bridge => _bridge;
+  Map<String, _ModuleRegistration> get moduleRegistrations =>
+      _moduleRegistrations;
+
+  set bridge(Bridge bridge) {
+    _bridge = bridge;
+    _bridge.apiCallReceived.listen(_handleApiCall);
+  }
+
+  void reset() {
+    _moduleRegistrations.clear();
+    _bridge = null;
+  }
+
+  void deregisterModule(SerializableModule module) {
+    if (module.serializableKey == null) return;
+    if (_moduleRegistrations.containsKey(module.serializableKey)) {
+      _moduleRegistrations[module.serializableKey].dispose();
+      _moduleRegistrations.remove(module.serializableKey);
+    }
+  }
+
+  void registerModule(SerializableModule module) {
+    if (module.serializableKey != null) {
+      final registration = new _ModuleRegistration(module);
+      _registerForLifecycleEvents(registration);
+      _moduleRegistrations[module.serializableKey] = registration;
+    } else {
+      _logger.warning('Unable to serialize module without serializableKey');
+    }
+  }
+
+  void _handleApiCall(Map apiCall) {
+    String module = apiCall['module'];
+    String method = apiCall['method'];
+    Object data = apiCall['data'];
+
+    SerializableModule targetModule = _moduleRegistrations[module].module;
+
+    if (targetModule != null) {
+      _deserializeAndCall(targetModule, method, data);
+    }
+  }
+
+  void _deserializeAndCall(
+      SerializableModule module, String method, List data) {
+    InstanceMirror apiMirror = reflect(module.api);
+
+    if (apiMirror != null) {
+      ClassMirror classMirror = apiMirror.type;
+
+      if (ClassMirror != null) {
+        MethodMirror apiMethodMirror =
+            classMirror.declarations[new Symbol(method)];
+
+        if (apiMethodMirror == null) {
+          _logger.warning(
+              'Method $method does not exist on ${module.serializableKey}\' module\'s API');
+          return;
```

Comment:
I like this approach of logging/returning early in the failure case instead of nested ifs, could you do this for the `apiMirror` and `classMirror` null checks above, too?